mmengine/model/base_module.py

```diff
@@ -100,14 +106,25 @@ def init_weights(self):
                     f'initialize {module_name} with init_cfg {self.init_cfg}',
                     logger=logger_name,
                     level=logging.DEBUG)
-                initialize(self, self.init_cfg)
+
+                init_cfgs = self.init_cfg
                 if isinstance(self.init_cfg, dict):
-                    # prevent the parameters of
-                    # the pre-trained model
-                    # from being overwritten by
-                    # the `init_weights`
-                    if self.init_cfg['type'] == 'Pretrained':
-                        return
+                    init_cfgs = [self.init_cfg]
+
+                # PretrainedInit have higher priority than any other init_cfg.
+                # Therefore we call initialize PretrainedInit last to overwrite
+                # the previous initialized weights.
+                # See details in https://github.com/open-mmlab/mmengine/issues/691 # noqa E501
+                other_cfgs = []
+                pretrained_cfg = []
+                for init_cfg in init_cfgs:
+                    assert isinstance(init_cfg, dict)
+                    if init_cfg.get('type') == 'Pretrained':
```

Comment:
Why does it use `get` method to obtain the type field rather than `init_cfg['type']`?